### PR TITLE
feat(shorebird_cli): add `shorebird build ipa`

### PR DESCRIPTION
**Pull request by felangel**
## Description

- feat(shorebird_cli): add `shorebird build ipa` (part of #518)

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/build/build_ipa_command.dart [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  String get name => 'ipa';
+
+  @override
+  Future<int> run() async {
+    if (!auth.isAuthenticated) {
+      printNeedsAuthInstructions();
+      return ExitCode.noUser.code;
+    }
+
+    final validationIssues = await runValidators();
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/commands/build/build_ipa_command.dart:54]: This is a fun wrinkle – the default validators list only contains `AndroidInternetPermissionValidator`, which doesn't apply here. I think we might need to have something similar in the app's entitlements file and/or its Info.plist.

File: packages/shorebird_cli/lib/src/commands/build/build_ipa_command.dart [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+  @override
+  String get description =>
+      '''Builds an .xcarchive and optionally .ipa for an iOS app to be generated for App Store submission.''';
+
+  @override
+  String get name => 'ipa';
+
+  @override
+  Future<int> run() async {
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/commands/build/build_ipa_command.dart:48]: Do we want to verify that the current OS can build ipas? I'm sure `flutter build ipa` will spit up appropriately so maybe we don't really need to.

File: packages/shorebird_cli/lib/src/commands/build/build_ipa_command.dart [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        codesign: codesign,
+      );
+    } on ProcessException catch (error) {
+      buildProgress.fail('Failed to build: ${error.message}');
+      return ExitCode.software.code;
+    }
+
+    buildProgress.complete();
+
+    const xcarchivePath = './build/ios/archive/Runner.xcarchive';
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/commands/build/build_ipa_command.dart:83]: Would probably be good to do a `p.join` here, even though this should work anywhere you can build an ipa

**Review by bryanoltman — APPROVED** [review]
LGTM, although we'll need to figure out validators at some point

**Review by bryanoltman — APPROVED** [review]
LGTM, although we'll need to figure out validators at some point

File: packages/shorebird_cli/lib/src/commands/build/build_ipa_command.dart (hunk 2 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/commands/build/build_ipa_command.dart:48]: Yeah I was planning to leverage the error handling built into the flutter tool but happy to adjust this

File: packages/shorebird_cli/lib/src/commands/build/build_ipa_command.dart (hunk 3 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/commands/build/build_ipa_command.dart:83]: Sounds good, will adjust in the next PR 👍

**Pull request merged**